components/molecule/notification/src/index.js

```diff
@@ -0,0 +1,162 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import Button from '@schibstedspain/sui-atom-button'
+import IconClose from '@schibstedspain/sui-svgiconset/lib/Close'
+import IconSuccess from '@schibstedspain/sui-svgiconset/lib/Linecheck'
+import IconError from '@schibstedspain/sui-svgiconset/lib/Lineerror'
+import IconInfo from '@schibstedspain/sui-svgiconset/lib/Lineinfo'
+import IconWarning from '@schibstedspain/sui-svgiconset/lib/Linewarning'
+import cx from 'classnames'
+
+const baseClass = 'sui-MoleculeNotification'
+
+const ICONS = {
+  info: <IconInfo svgClass={`${baseClass}-icon`} />,
+  error: <IconError svgClass={`${baseClass}-icon`} />,
+  success: <IconSuccess svgClass={`${baseClass}-icon`} />,
+  system: <IconInfo svgClass={`${baseClass}-icon`} />,
+  warning: <IconWarning svgClass={`${baseClass}-icon`} />
+}
+
+const AUTO_CLOSE_TIME = {
+  s: 3000,
+  m: 6000,
+  l: 9000,
+  manual: null
+}
+
+const TRANSITION_DELAY = 1000 // ms
+const TEXT_MAX_LENGHT = 110 // chars
+const BUTTONS_MAX = 3 // buttons
+
+class MoleculeNotification extends Component {
+  state = {
+    show: this.props.show,
+    delay: false
+  }
+
+  componentWillReceiveProps (nextProps) {
+    (this.state.show !== nextProps.show) && this.toggleShow()
+  }
+
+  shouldComponentUpdate (nextProps, nextState) {
+    const { show, delay } = this.state
+    return (show !== nextState.show) || (delay !== nextState.delay)
+  }
+
+  toggleShow = () => {
+    const show = !this.state.show
+    const { onClose, effect, autoClose } = this.props
+
+    effect
+      ? this.setState({ show, delay: true }, this.removeDelay(show))
+      : this.setState({ show })
+
+    if (show) {
+      const autoCloseTime = AUTO_CLOSE_TIME[autoClose]
+      autoCloseTime && this.autoClose(autoCloseTime)
+    } else {
+      clearTimeout(this.autoCloseTimout)
+      onClose && onClose()
```

Comment:
Instead creating this kind of check, I would create `onClose` with an empty function as a default Props.